source/beamline/CLS/CLSBeamlineStatus.h

```diff
@@ -42,6 +42,8 @@ class CLSBeamlineStatus : public CLSBiStateGroup
 	/// Returns the beam on value for the given control.
 	double componentBeamOnValue(AMControl *control) const { return controlState1ValueMap_.value(control, -1); }
 
+	/// Returns the beamline status PV
+	AMControl * beamlineStatusPVControl() { return beamlineStatusPVControl_ ? beamlineStatusPVControl_ : this; }
```

Comment:
SXRMB has a beamline status PV, which can tell us whether beam is ready or not. I think Vespers might have one as well.

If a beamline doesn't have the beam status PV, then we will use CLSBeamlineStatus as the PV, which will mark beam ready or not via checking the status of the shutters and valves